txpipe/twopoint.py

```diff
@@ -234,16 +240,72 @@ def write_output(self, data, meta, results):
 
         # Add the data points that we have one by one, recording which
         # tracer they each require
+        #ggs = []
+        #ngs = []
+        #nns = []
+        comb = []
         for d in results:
-            tracer1 = f'source_{d.i}' if d.corr_type in [XIP, XIM, GAMMAT] else f'lens_{d.i}'
-            tracer2 = f'source_{d.j}' if d.corr_type in [XIP, XIM] else f'lens_{d.j}'
+            #First lets try and get Covariance matrix
+            tracer1 = f'source_{d.i}' if d.corr_type in [XI, GAMMAT] else f'lens_{d.i}'
+            tracer2 = f'source_{d.j}' if d.corr_type in [XI] else f'lens_{d.j}'
+            comb.append(d.object)
+            if d.corr_type == XI:
```

Comment:
This been repeats itself needlessly - can you move things that are the same in each branch of the `if` statement outside it?

Reply:
Think This should be resolved, please check